metis-media-service/src/main/java/eu/europeana/metis/mediaprocessing/extraction/OEmbedProcessor.java

```diff
@@ -0,0 +1,232 @@
+  }
+
+  /**
+   * Is valid type photo boolean.
+   *
+   * @param oEmbedModel the o embed model
+   * @return the boolean
+   */
+  private static boolean isValidTypePhoto(OEmbedModel oEmbedModel) {
+    return oEmbedModel != null && oEmbedModel.getType() != null
+        && oEmbedModel.getType().equalsIgnoreCase("photo")
+        && oEmbedModel.getUrl()!=null && !oEmbedModel.getUrl().isEmpty()
+        && (oEmbedModel.getWidth() > 0 && oEmbedModel.getHeight() > 0);
+  }
+
+  /**
+   * Is valid type video boolean.
+   *
+   * @param oEmbedModel the o embed model
+   * @return the boolean
+   */
+  private static boolean isValidTypeVideo(OEmbedModel oEmbedModel) {
+    return oEmbedModel != null && oEmbedModel.getType() != null
+        && oEmbedModel.getType().equalsIgnoreCase("video")
+        && oEmbedModel.getHtml()!=null && !oEmbedModel.getHtml().isEmpty()
+        && (oEmbedModel.getWidth() > 0 && oEmbedModel.getHeight() > 0);
+  }
+
+  /**
+   * Has valid version boolean.
+   *
+   * @param oEmbedModel the o embed model
+   * @return the boolean
+   */
+  private static boolean hasValidVersion(OEmbedModel oEmbedModel) {
+    return oEmbedModel != null && oEmbedModel.getVersion() != null
+        && oEmbedModel.getVersion().startsWith("1.0");
+  }
+
+  /**
+   * Gets duration from model.
+   *
+   * @param oEmbedModel the o embed model
+   * @return the duration from model
+   */
+  private static Double getDurationFromModel(OEmbedModel oEmbedModel) {
+    Double duration;
+    try {
+      duration = Double.parseDouble(oEmbedModel.getDuration());
+    } catch (NumberFormatException e) {
+      duration = 0.0;
+    }
+    return duration;
+  }
+
+  /**
+   * Process a resource by extracting the metadata from the content.
+   *
+   * @param resource The resource to process. Note that the resource may not have content (see
+   * {@link MediaExtractorImpl#shouldDownloadForFullProcessing(String)}).
+   * @param detectedMimeType The mime type that was detected for this resource (may deviate from the mime type that was provided
+   * by the server and which is stored in {@link Resource#getProvidedMimeType()}).
+   * @param mainThumbnailAvailable Whether the main thumbnail for this record is available. This may influence the decision on
+   * whether to generate a thumbnail for this resource.
+   * @return The result of the processing.
+   * @throws MediaExtractionException In case something went wrong during the extraction.
+   */
+  @Override
+  public ResourceExtractionResult extractMetadata(Resource resource, String detectedMimeType, boolean mainThumbnailAvailable)
+      throws MediaExtractionException {
+
+    ResourceExtractionResult resourceExtractionResult;
+    // the content for this oembed needs to be downloaded to be examined
+    if (resource.getContentPath() != null) {
+      try {
+        OEmbedModel embedModel = null;
+        if (detectedMimeType.startsWith("application/json")) {
```

Comment:
I just realized, why are we not checking for the correct value? `application/json(+oembed)` and `application/xml(+oembed)`?
And why continue the code if the embedModel remains null after the check?

Reply:
because not all the oEmbed endpoints return `application/json+oembed` or `application/xml+oembed`.  the mimetype can also be a plain `application/json` or `application/xml` then the content is analysed. if `oembedModel` is valid is processed as it otherwise if it is null (meaning the oEmbed was not handled) then is passed to next processor that is the why the chain of processors.

Reply:
I'm reading this a bit more and I have questions. :D 
I think I see where the confusion comes from.
The spec apparently says it has to be exactly `application/json` or `text/xml`(and not startsWith. Check also the difference for xml). This is what is expected for the responses from the endpoint.
Additionally, it can be `application/json+oembed` or `text/xml+oembed` if it is a in a `link` tag which we don't use.

But the tika detector implemented "creates" the mime type by parsing the content basically modifying a potential mime-type returned from the response of the url.
So if it is overwritten in any case from the detector shouldn't the check here be precise,`application/json+oembed` or `application/xml+oembed`?

In theory we should be checking the returned mimetype of the response header, check that it is exactly `application/json` or `text/xml` and only then check if it parses to oembed. If conditions hold true we then modify the value by adding the `+oembed`
It is a bit confusing though because even the description of the ticket describes something different than the spec.
What holds true?

Reply:
and the examples from vimeo or youtube are then different, they don't follow the specification at all :see_no_evil:  when is an oembed xml response  mimetype is `application/xml` instead of `text/xml` :zany_face: 

For the tika processor, I made it explicitly to indicate that it was an oembed by returning `application/json+oembed` or `application/xml+oembed` and yes, if they conditions happens tika overwrites the mimetype, otherwise returns the octet stream, and continues to evaluate to the next detector.

Should we ask product owner what is the desired implementation?